### PR TITLE
feat(router): add routes to RouteConfigLoadEnd (#16282)

### PR DESCRIPTION
**Pull request by meeroslav**
**Please check if the PR fulfills these requirements**

* [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
* [X] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?**
When accessing modules lazily, RouteConfigLoadEnd gets fired but before routes are loaded and assigned to route._loadedConfig and therefore accessible.
More details in: [https://github.com/angular/angular/issues/16282](https://github.com/angular/angular/issues/16282)

**What is the new behavior?**
RouteConfigLoadEnd is fired once routes are fetched and has routes as a second parameter.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by meeroslav** [comment]
I signed it!

File: packages/router/src/router_config_loader.ts [hunk 1]
```diff
@@ -35,13 +35,14 @@ export class RouterConfigLoader {
     const moduleFactory$ = this.loadModuleFactory(route.loadChildren !);
 
     return map.call(moduleFactory$, (factory: NgModuleFactory<any>) => {
+      const module = factory.create(parentInjector);
```
Comment (vicb) [inline packages/router/src/router_config_loader.ts]: What we want to measure the the load time and here you incidentally add the module creation time.

Adding a new event type would be better IMO.

Comment (meeroslav) [inline packages/router/src/router_config_loader.ts]: I didn't know that part. It makes sense.

Should I propose a solution with new event type e.g. RouteChildrenLoaded and revert current changes or does this have to be further discussed within the team first?

**Comment by jasonaden** [comment]
@meeroslav Thanks for this PR and sorry you didn't get an answer earlier.

If we want to add this feature, I think you should add a new event. But can you first create an issue as a feature request so this could be discussed? Put in there what the feature is, but also the use case. It's easier to determine the best way to handle a problem if we know what the problem is that you're trying to solve.

I'll close this PR so you can open the issue and we can iterate there first.

**Comment by meeroslav** [comment]
@jasonaden The issue was already opened 3 months ago and mentioned [here](https://github.com/angular/angular/issues/16282) and [here](https://github.com/angular/angular/issues/14036). Also, package that depends on this functionality is `localize-router` with appropriate issue mentioned [here](https://github.com/Greentube/localize-router/issues/45).

Please let me know if you need more information.

If it's easier I can create new pull request based on @vicb recommendation.

**Pull request closed without merging**